### PR TITLE
Add C# examples to Using NavigationMaps

### PR DESCRIPTION
**Pull request by ShawnHardern**
Add C# examples to Using NavigationMaps

### CODE REVIEW

**Review by smix8 — APPROVED** [review]

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 1]
```diff
@@ -33,13 +33,33 @@ The 3D default navigation map RID can be obtained with ``get_world_3d().get_navi
     func _ready() -> void:
         var default_navigation_map_rid: RID = get_world_2d().get_navigation_map()
 
+ .. code-tab:: csharp 2D C#
+
+    public partial class MyClass : Node2D
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:38]: We usually follow the pattern `My{BaseClass}` for naming classes in the docs, unless the GDScript version names them explicitly with `class_name`.

```suggestion
    public partial class MyNode2D : Node2D
```

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 2]
```diff
@@ -33,13 +33,33 @@ The 3D default navigation map RID can be obtained with ``get_world_3d().get_navi
...
  .. code-tab:: gdscript 3D GDScript
 
     extends Node3D
 
     func _ready() -> void:
         var default_navigation_map_rid: RID = get_world_3d().get_navigation_map()
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : Node3D
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:55]:
```suggestion
    public partial class MyNode3D : Node3D
```

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 3]
```diff
@@ -66,6 +86,17 @@ Navigation regions and avoidance agents can only be part of a single navigation
         var new_navigation_map: RID = NavigationServer2D.map_create()
         NavigationServer2D.map_set_active(new_navigation_map, true)
 
+ .. code-tab:: csharp 2D C#
+
+    public partial class MyClass : Node2D
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:91]:
```suggestion
    public partial class MyNode2D : Node2D
```

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 4]
```diff
@@ -74,6 +105,17 @@ Navigation regions and avoidance agents can only be part of a single navigation
         var new_navigation_map: RID = NavigationServer3D.map_create()
         NavigationServer3D.map_set_active(new_navigation_map, true)
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : Node3D
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:110]:
```suggestion
    public partial class MyNode3D : Node3D
```

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 5]
```diff
@@ -33,13 +33,33 @@ The 3D default navigation map RID can be obtained with ``get_world_3d().get_navi
     func _ready() -> void:
         var default_navigation_map_rid: RID = get_world_2d().get_navigation_map()
 
+ .. code-tab:: csharp 2D C#
+
+    public partial class MyClass : Node2D
+    {
+        public override void _Ready()
+        {
+            Rid defaultNavigationMapRid = GetWorld2D().GetNavigationMap();
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:42]: Prefer properties to hidden methods.

```suggestion
            Rid defaultNavigationMapRid = GetWorld2D().NavigationMap;
```

File: tutorials/navigation/navigation_using_navigationmaps.rst [hunk 6]
```diff
@@ -33,13 +33,33 @@ The 3D default navigation map RID can be obtained with ``get_world_3d().get_navi
...
     func _ready() -> void:
         var default_navigation_map_rid: RID = get_world_3d().get_navigation_map()
 
+ .. code-tab:: csharp 3D C#
+
+    public partial class MyClass : Node3D
+    {
+        public override void _Ready()
+        {
+            Rid defaultNavigationMapRid = GetWorld3D().GetNavigationMap();
```
Comment (raulsntos) [inline tutorials/navigation/navigation_using_navigationmaps.rst:59]:
```suggestion
            Rid defaultNavigationMapRid = GetWorld3D().NavigationMap;
```

**Review by raulsntos — CHANGES_REQUESTED** [review]

**Review by raulsntos — APPROVED** [review]

**Comment by AThousandShips** [comment]
Thanks! And congratulations on your first merged contribution! 🎉

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10038.

**Pull request merged**